tests/components/flo/test_device.py

```diff
@@ -83,18 +82,47 @@ async def test_device(
 
     call_count = aioclient_mock.call_count
 
-    async_fire_time_changed(hass, dt_util.utcnow() + timedelta(seconds=90))
+    freezer.tick(timedelta(seconds=90))
+    async_fire_time_changed(hass)
     await hass.async_block_till_done()
 
     assert aioclient_mock.call_count == call_count + 6
 
-    # test error sending device ping
+
+async def test_device_failures(
+    hass: HomeAssistant,
+    config_entry,
+    aioclient_mock_fixture,
+    aioclient_mock: AiohttpClientMocker,
+    freezer: FrozenDateTimeFactory,
+) -> None:
+    """Test Flo by Moen devices buffer API failures."""
+    config_entry.add_to_hass(hass)
+    assert await hass.config_entries.async_setup(config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    def assert_state(state: str) -> None:
+        assert (
+            hass.states.get("sensor.smart_water_shutoff_current_system_mode").state
+            == state
+        )
+
+    assert_state("home")
+
+    async def move_time_and_assert_state(state: str) -> None:
+        freezer.tick(timedelta(seconds=65))
+        async_fire_time_changed(hass)
+        await hass.async_block_till_done()
+        assert_state(state)
+
+    aioclient_mock.clear_requests()
     with patch(
         "homeassistant.components.flo.device.FloDeviceDataUpdateCoordinator.send_presence_ping",
```

Comment:
We shouldn't patch the coordinator method but the library API client method.

https://github.com/home-assistant/core/blob/2c0b8976586597646e91629575c38d255e57562f/homeassistant/components/flo/device.py#L197

Reply:
That’s how I had it…

Reply:
I think the correct target is `"homeassistant.components.flo.async_get_api"`.

Reply:
Although that would require mocking the user info return value too. Just patching the client ping method is probably OK too.

Reply:
I’ll open another PR later to change the target to the clients ping method like the reference above.